cli/src/git_util.rs

```diff
@@ -172,7 +172,45 @@ pub fn with_remote_git_callbacks<T>(
     f(callbacks)
 }
 
-pub fn print_git_import_stats(ui: &mut Ui, stats: &GitImportStats) -> Result<(), CommandError> {
+pub fn print_git_import_stats(
+    ui: &mut Ui,
+    stats: &GitImportStats,
+    show_ref_stats: bool,
+) -> Result<(), CommandError> {
+    use std::collections::BTreeMap;
+
+    if show_ref_stats && !stats.new_remote_refs.is_empty() {
+        let mut new_remote_branches: BTreeMap<_, Vec<_>> = BTreeMap::new();
+        let mut tags = Vec::new();
+        let mut local_branches = Vec::new();
+        for refname in &stats.new_remote_refs {
+            match refname {
+                RefName::RemoteBranch { branch, remote } => {
+                    new_remote_branches.entry(remote).or_default().push(branch);
+                }
+                RefName::Tag(tag) => tags.push(tag),
+                RefName::LocalBranch(b) => local_branches.push(b),
+            }
+        }
+        local_branches.sort_unstable();
+        for branch in local_branches {
+            writeln!(ui.stderr(), "* [imported local branch]  {branch}")?;
+        }
+        for (remote, branches) in new_remote_branches.iter_mut() {
+            writeln!(ui.stderr(), "From {remote}")?;
+            branches.sort_unstable();
+            for branch in branches {
+                writeln!(ui.stderr(), "* [new untracked branch]  {branch}")?;
```

Comment:
"new" and "untracked" are wrong.

The ref might be just moved or deleted. To check if it's moved/created/deleted, `(old_remote_ref, new_target)` pair can be used.

To check if it's untracked/tracked, we'll probably need to query `repo.view().get_remote_branch(branch, remote).is_tracking()`.

Reply:
Partially resolved, see
https://github.com/martinvonz/jj/pull/3044/files#diff-9bc1bb41a6477444bc8efccd5a0c7b053997c4aa67a216e87f8ee819bedbc7acR239-R245

Reply:
Resolved